### DISCUSSION TITLE
Remove notification related to a repository when un-watching it

### DISCUSSION THREAD

**Discussion by yguedidi** [community/community#63958] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

# Context

It's the second time I face an apparently known bug that creates "ghost" notifications: on the Notifications page, in the list of repositories a repository appears, but when selecting it to see its notifications, an empty table appears, even if it shows there is one unread notification.

# Workaround

The GitHub support team suggested a hack to mark all notifications as read via the API, but this will only mark the ghost notification as read, not as done.

# Issue

The repository will continue to appear in the list... And according to the support team it will disappear after 4 months because it seems that read notifications older than 4 months are not listed anymore in the Notifications page.
I confirm this works, but it means having that repository listed for 4 months...
It's frustrating and not practical to have that repository persist with a ghost read notification that we can't mark as done neither via the interface neither via the API

# Proposal

As a workaround but quite logical to me, I'd suggest to simply remove all notifications related to a repository when un-watching it.

**Comment by tbroadley** [comment]
The GitHub API now has an endpoint that will let you mark the ghost notification as done!

This marks your first notification as done:

```
NOTIFICATION_ID=$(gh api /notifications | jq '.0.id | tonumber')
gh api -X DELETE /notifications/threads/$NOTIFICATION_ID
```

**Reply by bugadani** [reply]
Thank you, you're a life saver

**Reply by bernard-wagner** [reply]
This worked for me:
```
gh api -X DELETE /notifications/threads/$(gh api /notifications | jq '.[0].id | tonumber')
```

**Reply by TomasVotruba** [reply]
@bernard-wagner Thank you :+1:  I was getting crazy, on how to read that notification :D it works well

**Reply by reslear** [reply]
@bernard-wagner thanks

**Reply by rajsite** [reply]
thanks, had a notification for a repo I can't even see (`plasmanotification` org?) and was driving me nuts

**Reply by B-3000** [reply]
Unfortunately, this is a known bug in GitHub notifications and hopefully they'll solve it soon.

But in the meantime, here are some of the other solutions found in  thread discussions regarding this -
1. https://github.com/orgs/community/discussions/6874#discussioncomment-1775744
2. https://github.com/orgs/community/discussions/6874#discussioncomment-14507162
3. https://gist.github.com/Nriver/42feddd2a369a8f1519b47adb81ebef8
4. https://github.com/emmanuel-ferdman/gh-gonest
5. https://gist.github.com/jeremystretch/2c09f76837fc5af787fe9ff7747ecf3f
6. https://github.com/orgs/community/discussions/174310#discussioncomment-14474599
7. https://github.com/orgs/community/discussions/6874#discussioncomment-14481926
8. https://gist.github.com/eseiler/d3a6f978ceea1853f0c558879cf859c1
